### PR TITLE
WIP: Add cost saving component

### PR DESCRIPTION
**Pull request by RobinKnipe**
Add a simple component that displays a cost saving amount.

### CODE REVIEW

File: apps/report/src/common/pages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import { PageProps } from '@not-govuk/app-composer';
+import { Form, before, after, past, minLength, required } from '@not-govuk/components';
+
+const Page: FC<PageProps> = props => (
+  <div className="govuk-grid-row">
+    <div className="govuk-grid-column-two-thirds">
+      <h1>Home</h1>
+      <p>This is the home page updated.</p>
+	<Form action="/result" method="get">
+  <Form.TextInput
```
Comment (daniel-ac-martin) [inline apps/report/src/common/pages/index.tsx:11]: Would be nice to fix this indentation.

Comment (daniel-ac-martin) [reply apps/report/src/common/pages/index.tsx:11]: In fact, this whole form looks like a test.
Do you want to commit it for now?

File: components/cost-saving-counter/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+HO LEV - Cost Saving Counter
+===
```
Comment (daniel-ac-martin) [inline components/cost-saving-counter/README.md:2]: I generally tidy this up a bit. Something like:
```md
LEV - Cost saving counter
=========================
```

File: components/cost-saving-counter/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+Then use it in your code as follows:
+
+```js
+import React, { createElement as h } from 'react';
+import CostSavingCounter from '@ho-lev/cost-saving-counter';
+
+export const MyComponent = props => (
+  <CostSavingCounter
+    // WRITEME
```
Comment (daniel-ac-martin) [inline components/cost-saving-counter/README.md:24]: I'd normally copy and paste in one of my stories here.

File: components/cost-saving-counter/assets/CostSavingCounter.scss [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  text-align: center;
+  border: 8px solid #9325b2;
+  border-radius: 7px;
+  margin-bottom: 1vh;
+
+  & > * {
+    color: #9325b2;
+    max-width: 100%;
+  }
+  &__heading {
```
Comment (daniel-ac-martin) [inline components/cost-saving-counter/assets/CostSavingCounter.scss:12]: Would be nice to have a blank line above.

File: components/cost-saving-counter/spec/CostSavingCounter.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import toJson from 'enzyme-to-json';
+
+describe('CostSavingCounter', () => {
+  describe('when given valid props', () => {
+    const group = 'the public';
+    const component = mount(h(CostSavingCounter, { costsaving: 2345.56, group: group }));
+
+    it('renders cost saving', () => expect(component.find('h2').text()).toEqual('£2,345.56'));
+    it('renders the group', ()=> expect(component.find('p').text()).toContain(group));
+    it('contains no regressions', () => expect(toJson(component)).toMatchSnapshot());
```
Comment (daniel-ac-martin) [inline components/cost-saving-counter/spec/CostSavingCounter.ts:13]: I don't think we should do snapshot testing like this.

I'd rather rely on Chromatic for visual regressions. (Of course there's an outstanding issue to fix that.)

File: components/cost-saving-counter/spec/__snapshots__/CostSavingCounter.ts.snap [hunk 6]
```diff
@@ -0,0 +1,22 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (daniel-ac-martin) [inline components/cost-saving-counter/spec/__snapshots__/CostSavingCounter.ts.snap:1]: This file should go as well.

File: components/cost-saving-counter/spec/CostSavingCounter.ts (hunk 5 above)
Comment (daniel-ac-martin) [reply components/cost-saving-counter/spec/CostSavingCounter.ts:13]: Storybook also offers built in support for this if we did want to do it. But I think it's worth reading the note at the bottom of the following page that sort of warns against it: https://storybook.js.org/docs/react/workflows/snapshot-testing

**Comment by daniel-ac-martin** [comment]
Looks okay but I wonder if the report app should be removed for now. - It doesn't look like it's adding any value.

File: components/cost-saving-counter/spec/CostSavingCounter.ts (hunk 5 above)
Comment (RobinKnipe) [reply components/cost-saving-counter/spec/CostSavingCounter.ts:13]: Yes, Chromatic definitely has some good advantages! I know you were hitting the free snapshot limit - do you think a 2-tiered approach might help? First stage: use `storyshot` to suggest some attention maybe needed (before push)... second stage: visual regressions and PR. @daniel-ac-martin

**Comment by RobinKnipe** [comment]
I'm going to redo this work with a [slightly] different approach, so I'll close this PR...

**Pull request closed without merging**